dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/step/UploadStep.java

```diff
@@ -87,8 +98,28 @@ public void doPatchProcessing(Context context, Request currentRequest, InProgres
             }
         }
         PatchOperation<?> patchOperation = new PatchOperationFactory().instanceOf(instance, op.getOp());
-        patchOperation.perform(context, currentRequest, source, op);
+        if (instance.equals(AbstractRestProcessingStep.UPLOAD_STEP_METADATA_OPERATION_ENTRY)) {
+            DCInputSet inputConfig = inputReader.getInputsByFormName(UploadStep.UPLOAD_STEP_METADATA_SECTION);
+            String[] split = patchOperation.getAbsolutePath(op.getPath()).split("/");
+            String metadata = findMetadata(split);
+            if (inputConfig.isFieldPresent(metadata)) {
+                patchOperation.perform(context, currentRequest, source, op);
+            } else {
+                throw new UnprocessableEntityException("The attribute " + metadata + " does not present in section "
+                                                                        + UploadStep.UPLOAD_STEP_METADATA_SECTION);
+            }
+        } else {
+            patchOperation.perform(context, currentRequest, source, op);
+        }
+    }
 
+    private String findMetadata(String[] metadata) {
+        for (String s : metadata) {
+            if (s.contains("dc.")) {
+                return s;
```

Comment:
I'm confused by this method.  Why is it only returning metadata fields in the "dc" schema?  I think this needs code comments or JavaDocs to explain why this is needed.

Reply:
I haven't noted that, yes it needs to be rewritten better. @Micheleboychuk I guess that you have introduced that to locate the metadata in the path string of the json patch operation. You can do that easily as the path must be /sections/uploads/files/<idx>/metadata/xxxxxx/... so if the split has a length >= 6 the split[5] will contains the metadata. Anyway, we also need to check that the json path is valid, i.e. /sections/uploads/files/<idx>/not-existing-prop/* should be rejected as well